src/ui/digital-exchange/components/SearchBar.js

```diff
@@ -0,0 +1,84 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import { Button, Icon } from 'patternfly-react';
+import { throttle } from '@entando/utils';
+
+class SearchBar extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      value: '',
+      filled: false,
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleKeyUp = this.handleKeyUp.bind(this);
+    this.handleClear = this.handleClear.bind(this);
+  }
+
+  handleChange(e) {
+    this.setState({
+      value: e.target.value,
+      filled: (e.target.value !== ''),
+    });
+  }
+
+  handleClear() {
+    const triggerSubmit = this.state.value !== '';
+    this.setState({
+      value: '',
+      filled: false,
+    });
+    if (triggerSubmit) {
+      throttle(() => this.handleSubmit());
+    }
+  }
+
+  handleKeyUp(e) {
+    if (e.key === 'Enter') this.handleSubmit();
+  }
+
+  handleSubmit() {
+    this.props.onSubmit(this.state.value);
+  }
+
+  render() {
+    return (
+      <div className="SearchBar__container">
+        <div className="SearchBar__textbox">
+          <input
+            type="text"
+            className="SearchBar__textbox--base"
+            onBlur={this.handleBlur}
+            onChange={this.handleChange}
+            onFocus={this.handleFocus}
+            onKeyUp={this.handleKeyUp}
```

Comment:
wrapping this in a form and using redux form makes most of this redundant probably

Reply:
okay i can change that. thanks for the input @nicpuddu